atlante/template/remote/remote.go

```diff
@@ -0,0 +1,77 @@
+package remote
+
+import (
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-spatial/maptoolkit/atlante/filestore"
+	"github.com/go-spatial/maptoolkit/atlante/internal/urlutil"
+)
+
+var RemoteDir = "remoteasset"
+
+type remoteMeta struct {
+	Filename string    `json:"filename"`
+	ETag     string    `json:"etag"`
+	URL      string    `json:"url"`
+	Date     time.Time `json:"timestamp"`
+}
+
+func attemptWrite(w filestore.FileWriter, name string, data []byte) (n int, err error) {
+	writer, err := w.Writer(name, true)
+	if err != nil {
+		return 0, err
+	}
+	return writer.Write(data)
+}
+
+func Remote(location string, fswriter filestore.FileWriter, useCached bool) (string, error) {
+	loc, err := url.Parse(location)
+	if err != nil {
+		return "", err
+	}
+	hostname := strings.ToLower(loc.Hostname())
+	normalizeURL := fmt.Sprintf("%v://%v/%v?%v#%v", loc.Scheme, hostname, loc.RawPath, loc.RawQuery, loc.Fragment)
```

Comment:
I think it would be better to use the [url.String()](https://golang.org/pkg/net/url/#URL.String) method here. All you're doing is lowercasing the hostname, which you can do by updating the `loc.Host` value